bin/compare-factory-config.py

```diff
@@ -0,0 +1,180 @@
+
+
+def run(argv):
+
+    # dictionary that adds GLIDEIN_ResourceNames under corresponding tags
+    topologyDB = {'resources': set(),
+                  'sites': set(),
+                  'facilities': set(),
+                  'resourceGroups': set()}
+    getTopologyData(topologyDB)
+    # cloning gfactory repository to a temporary directory
+    tempDir = tempfile.mkdtemp()
+    git.Repo.clone_from(
+        'https://github.com/opensciencegrid/osg-gfactory',
+        to_path=tempDir
+    )
+    gfactory = []
+    gfactory.extend(glob.glob(os.path.abspath(tempDir) + '/*.xml')
+                    + (glob.glob(os.path.abspath(tempDir) + '/OSG_autoconf/*.yml')))
+    """
+    Hardcoded xml sources
+    gfactory = [
+        'https://raw.githubusercontent.com/opensciencegrid/osg-gfactory/master/10-cms-cern_osg.xml',
+        'https://raw.githubusercontent.com/opensciencegrid/osg-gfactory/master/10-cmsonly-cern_fnal.xml',
+        'https://raw.githubusercontent.com/opensciencegrid/osg-gfactory/master/10-cmsopp-cern_osg.xml',
+        'https://raw.githubusercontent.com/opensciencegrid/osg-gfactory/master/10-cmst1-all.xml',
+        'https://raw.githubusercontent.com/opensciencegrid/osg-gfactory/master/10-cmst1-uscmst2-all.xml',
+        'https://raw.githubusercontent.com/opensciencegrid/osg-gfactory/master/10-covid19-osg.xml',
+        'https://raw.githubusercontent.com/opensciencegrid/osg-gfactory/master/10-cream.xml',
+        'https://raw.githubusercontent.com/opensciencegrid/osg-gfactory/master/10-fermi-fnal_osg.xml',
+        'https://raw.githubusercontent.com/opensciencegrid/osg-gfactory/master/10-noncms-osg.xml',
+        'https://raw.githubusercontent.com/opensciencegrid/osg-gfactory/master/10-testonly-itb.xml',
+        'https://raw.githubusercontent.com/opensciencegrid/osg-gfactory/master/20-local-itb.xml',
+        'https://raw.githubusercontent.com/opensciencegrid/osg-gfactory/master/30-local-cern.xml',
+        'https://raw.githubusercontent.com/opensciencegrid/osg-gfactory/master/30-local-fnal.xml'
+    ] """
+    # dictionary that stores (GLIDEIN_ResourceNames: entry name) pairs
+    gfactoryDB = {}
+    for xml in gfactory:
+        getGfactoryData(gfactoryDB, xml)
+
+    # comparing gfactory with Topology resources
+    # GLIDEIN_ResourceNames that does not match resources records in TopologyDB
+    nonMatchNames = set(gfactoryDB.keys()).difference(
+        topologyDB['resources'])
+
+    # corresponding entry names of those nonmatches
+    nonMatchEntries = [gfactoryDB[x] for x in nonMatchNames]
+    print(f'\nEntries that does not have a record in Topology resources: \n\n',
+          sorted(nonMatchEntries), '\n')
+
+    # The nonmatching GLIDEIN_ResourceNames that has a record in TopologyDB
+    matchedEntries = findMatches(nonMatchNames, topologyDB, gfactoryDB)
+    print(f'\nEntries that does not have a record in Topology resources tag but have records in Topology database: \n\n',
+          sorted(matchedEntries), '\n')
```

Comment:
I'd like to see the format of this output, one entry per line:

```
- <Factory Entry Name>: <Topology resource group, site, or facility>
```

Replacing `<Factory Entry Name>` with the entry name (e.g. [CLAS12_FR_GRIF_ipngrid06](https://github.com/opensciencegrid/osg-gfactory/blob/master/10-noncms-osg.xml#L3)) and `<Topology resource group, site, and/or facility>` with any topology names that `GLIDEIN_ResourceName` matches